source/module_io/test/write_input_test.cpp

```diff
@@ -0,0 +1,687 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#define private public
+#include "module_io/input.h"
+/************************************************
+ *  unit test of write_input.cpp
+ ***********************************************/
+
+/**
+ * - Tested Functions:
+ *   - Print()
+ *     - output the information in the input file.
+ */
+class write_input : public testing::Test
+{
+     protected:
+     Input INPUT;
+};
+
+TEST_F(write_input,print)
+{
+    std::string output;
+    INPUT.Read("./support/witestfile");
+    std::string output_file = "write_input_test.log";
+    INPUT.Print(output_file);
+    int a=access("write_input_test.log",00);
+    EXPECT_EQ(a,0);
+    std::ifstream ifs("write_input_test.log");
+    getline(ifs,output);
+    EXPECT_THAT(output,testing::HasSubstr("INPUT_PARAMETERS"));
+    getline(ifs,output);
```

Comment:
the multiple getline codes can be avoided by using std::istreambuf_iterator. See example in module_base/test/global_function_test.cpp: line 284-292

Reply:
> 



> the multiple getline codes can be avoided by using std::istreambuf_iterator. See example in module_base/test/global_function_test.cpp: line 284-292

OK,I have updated the code.